lib/sparrow/h2_worker.ex

```diff
@@ -72,6 +90,14 @@ defmodule Sparrow.H2Worker do
       Logger.info(fn ->
         "action=terminate, reason=#{inspect(reason)}, connection_ref!=nil"
       end)
+
+    :telemetry.execute(
+      [:sparrow, :h2_worker, :terminate],
+      %{reason: reason},
+      %{state: state}
```

Comment:
I think leaking the whole state to telemetry may be too much. Let's return data that may be useful on a higher level, like name of the worker, type of the worker (apns vs fcm), reason of crash).